lib/quby/table_backend/disk_table.rb

```diff
@@ -0,0 +1,132 @@
+require 'quby/table_backend/table_dimension'
+module Quby::TableBackend
+  class DiskTable
+    require "pathname"
+
+    def self.disk_table_root
+      fail 'Quby.lookup_table_path not configured' if Quby.lookup_table_path.blank?
+      Quby.lookup_table_path
+    end
+
+    def self.validate_jailed_path(path)
+      full_path = File.expand_path(path)
+      jail = File.expand_path(disk_table_root)
+      full_path.starts_with?(jail)
+    end
+
+    # range should either be:
+    #  an empty array, which returns a range that will accept any value on an 'include?' call
+    #  a 2 long float string array, these will be converted into a float range
+    #    float ranges will be open ended on the right side
+    #  a varied length non-float string array, to signify a set of categorical options (like 'male', 'female', 'other')
+    #    this will be returned as is
+    def self.parse_range(range)
+      if range.length == 0
+        Quby::TableBackend::TableDimension::AcceptsAnythingRange.instance
+      elsif parse_float(range.first).present?
+        parse_float_range(range)
+      else
+        range
+      end
+    end
+
+    def self.parse_float_range(range)
+      maybe_float_start = parse_float(range[0])
+      maybe_float_end = parse_float(range[1])
+      if maybe_float_end.present? && range.length == 2
+        return (maybe_float_start...maybe_float_end)
+      else
+        fail "#{range} has a float start but not a float end"
+      end
+    end
+
+    def self.parse_float(maybe_float_string)
+      float = Float(maybe_float_string) rescue nil
+      case maybe_float_string
+      when 'minfinity' then -Float::INFINITY
+      when 'infinity' then Float::INFINITY
+      else float
+      end
+    end
+
+    def initialize(key)
```

Comment:
initialize verwacht ik altijd als eerste method (als daar niet eentje is, ga ik er vanuit dat er niet een is.